### ISSUE TITLE
Getting Invalid status code 500 with createTestAccount()

### ISSUE THREAD

**Issue by angelod1as** [nodemailer/nodemailer#1140]
Please fill the following questionnaire about your issue:

1. What kind of issue are you reporting?
- [ ] A bug in Nodemailer
- [ ] A bug in a plugin of Nodemailer (eg. issues with nodemailer-sendgrid)
- [ ] Feature request
- [X] Looking for help to resolve some kind of problem with Nodemailer

2. Are you listed as a sponsor of Nodemailer project (see Sponsors button above)?
- [ ] Yes. Sponsors get priority support
- [X] No. Unless it is a bug in Nodemailer you might find support from public forums like [StackOverflow](https://stackoverflow.com/search?q=nodemailer).

3. State your problem here:

I'm using default code from Nodemailer page (console added by me):

```
nodemailer.createTestAccount((err, account) => {
    console.log(err);
    // create reusable transporter object using the default SMTP transport
    let transporter = nodemailer.createTransport({
        host: 'smtp.ethereal.email',
        port: 587,
        secure: false, // true for 465, false for other ports
        auth: {
            user: account.user, // generated ethereal user
            pass: account.pass  // generated ethereal password
        }
    });
});
```

It returns me the following error:

```
Error: Invalid status code 500
    at ClientRequest.<anonymous> (/Users/angelodias/GIT/bootcamp/04-iniciando-back-end/node_modules/nodemailer/lib/fetch/index.js:221:23)
    at ClientRequest.emit (events.js:223:5)
    at ClientRequest.EventEmitter.emit (domain.js:475:20)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:583:27)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:115:17)
    at TLSSocket.socketOnData (_http_client.js:456:22)
    at TLSSocket.emit (events.js:223:5)
    at TLSSocket.EventEmitter.emit (domain.js:475:20)
    at addChunk (_stream_readable.js:309:12)
    at readableAddChunk (_stream_readable.js:290:11) {
  type: 'FETCH',
  sourceUrl: 'https://api.nodemailer.com/user'
}
```

What am I doing wrong? Is there any bug?

I tried with .then().catch() and it catches the same error.

Running Node v12.14.1 and Nodemailer 6.4.6

**Comment by goyney** [comment]
I would like to chime in here that we are also now seeing this issue. It seems to have just started happening within the past 24 hours or so. Every time we try to create a test account, we get a 500 error.

Here's our helper function to generate the account:
```js
const createTestSmtpAccount = () => {
  return new Promise((resolve, reject) => {
    nodemailer.createTestAccount((error, account) => {
      if (error) {
        return reject(`Failed to create a testing account. Details: ${error.message}`);
      }
      return resolve({
        host: account.smtp.host,
        password: account.pass,
        port: account.smtp.port,
        secure: account.smtp.secure,
        username: account.user
      });
    });
  });
};
```

Using that always returns:
```
Error: Invalid status code 500
    at ClientRequest.req.on.res (node_modules/nodemailer/lib/fetch/index.js:221:23)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:565:21)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:111:17)
    at TLSSocket.socketOnData (_http_client.js:451:20)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at TLSSocket.Readable.push (_stream_readable.js:224:10)
    at TLSWrap.onStreamRead (internal/stream_base_commons.js:94:17)
```

The supplied code has been in use for quite some time and this just started happening, as I said, probably within the last 24 hours or so. We've done some intensive digging to make sure our enviornment hasn't changed, but found nothing. Additionally, running it locally gives the same output.

We also checked to see if Ethereal was having issues, but they have not reported anything on their status page: https://status.ethereal.email/

**Comment by goyney** [comment]
It looks like there were some changes to Wildduck around the `/users` API endpoint about a month ago. Potential correlation?

https://github.com/nodemailer/wildduck/blob/master/lib/api/users.js

**Comment by andris9** [comment]
Fixed now. One service stopped working because of out of memory.

**Comment by stephenstuder** [comment]
@andris9 Is this happening again? Same issues as above happening to me currently.

**Comment by judsonjuniorr** [comment]
> @andris9 Is this happening again? Same issues as above happening to me currently.

Happening here too

**Comment by stephenstuder** [comment]
Opened a new issue with reference to this thread since others seem to be experiencing it as well.

**Comment by AnchorNoa** [comment]
> @andris9 Is this happening again? Same issues as above happening to me currently.

Having the same issue.

**Comment by andris9** [comment]
See https://github.com/nodemailer/nodemailer/issues/1333#issuecomment-932819586

**Comment by dralos** [comment]
i started having the same issue today

**Comment by andris9** [comment]
Tl;dr I tried to save on costs by downgrading the DB server. Everything worked well until oomkiller started to take down that database process 🤷‍♂️ Restored previous DB setup so it should work better now.

**Comment by tiagoluzs** [comment]
Check if you have this line in your code:

`let testAccount = await nodemailer.createTestAccount();`

Remove it and it will fix your issue.

**Comment by mriecken** [comment]
Time to reboot the server again?

**Comment by venjo47** [comment]
issue is still persist in smtp.gmail.com, anyone solve the issue ?

**Comment by eichlerz** [comment]
please reboot :D

**Comment by andris9** [comment]
I rebooted the server, and it should work again. If you are sending actual mail to Gmail and you get errors for Ethereal.email, you're doing it wrong.

**Comment by eichlerz** [comment]
It seems the same issue is back again :(

**Comment by bruno-silva5** [comment]
Same problem here :/

**Comment by adamcole123** [comment]
I'm getting this issue

**Comment by alagappan-qa** [comment]
Started getting this issue again

**Comment by mathare** [comment]
Just started getting this issue again

**Comment by adityasirohi-philips** [comment]
Facing this issue since last 1 hour. What is the resolution?

**Comment by shenyusun** [comment]
Getting the same issue here.

**Comment by mtheusbrito** [comment]
I'm having the same problem here.

**Comment by gautam351** [comment]
Having the same issue ....getting  "Invalid status code 500" .

**Comment by andris9** [comment]
Unfortunately, this is a recurring issue. The server that runs [ethereal.email](https://ethereal.email/) runs out of memory now and then and I have to manually fix this.

As an alternative for Ethereal.email I would suggest using [Mailtrap](https://mailtrap.io/). It is better suited for any production use.

**Comment by akim3235** [comment]
Getting the same issue here. Please fix.

**Comment by yoannmaingon** [comment]
Getting the same issue here. Please fix. :)
